### PR TITLE
Implemented and activated Identity module

### PR DESCRIPTION
**Pull request by markopoloparadox**
# Overview
A federated naming system, allowing for multiple registrars to be added from a specified origin. Registrars can set a fee to provide identity-verification service. Anyone can put forth a proposed identity for a fixed deposit and ask for review by any number of registrars (paying each of their fees). Registrar judgements are given as an enum, allowing for sophisticated, multi-tier opinions.

Some judgements are identified as sticky, which means they cannot be removed except by complete removal of the identity, or by the registrar. Judgements are allowed to represent a portion of funds that have been reserved for the registrar.

A super-user can remove accounts and in doing so, slash the deposit.

All accounts may also have a limited number of sub-accounts which may be specified by the owner; by definition, these have equivalent ownership and each has an individual name.

The number of registrars should be limited, and the deposit made sufficiently large, to ensure no state-bloat attack is viable.

# Interface
**Dispatchable Functions**
**For general users**
- set_identity - Set the associated identity of an account; a small deposit is reserved if not already taken.
- clear_identity - Remove an account's associated identity; the deposit is returned.
- request_judgement - Request a judgement from a registrar, paying a fee.
- cancel_request - Cancel the previous request for a judgement.

**For general users with sub-identities**
- set_subs - Set the sub-accounts of an identity.
- add_sub - Add a sub-identity to an identity.
- remove_sub - Remove a sub-identity of an identity.
- rename_sub - Rename a sub-identity of an identity.
- quit_sub - Remove a sub-identity of an identity (called by the sub-identity).

**For registrars**
- set_fee - Set the fee required to be paid for a judgement to be given by the registrar.
- set_fields - Set the fields that a registrar cares about in their judgements.
- provide_judgement - Provide a judgement to an identity.

**For super-users**
- add_registrar - Add a new registrar to the system.
- kill_identity - Forcibly remove the associated identity; the deposit is lost.

# References
Parent issue: https://github.com/availproject/engineering/issues/84

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -1016,6 +1048,8 @@ construct_runtime!(
 		Multisig: pallet_multisig = 34,
 		VoterList: pallet_bags_list::<Instance1> = 35,
 		NominationPools: pallet_nomination_pools = 36,
+
```
Comment (markopoloparadox) [inline runtime/src/lib.rs:1051]: TODO Remove empty line

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -15,8 +15,10 @@ targets = ["x86_64-unknown-linux-gnu"]
 [dependencies]
 
 # Internal
-avail-core = { version = "0.5", default-features = false, features = ["runtime"] }
-kate = { version = "0.8" , default-features = false }
+avail-core = { version = "0.5", default-features = false, features = [
```
Comment (markopoloparadox) [inline runtime/Cargo.toml:18]: TODO Reverent toml formatting

**Review by delroybosco — COMMENTED** [review]
Have we tested this internally? @markopoloparadox

**Comment by markopoloparadox** [comment]
> Have we tested this internally? @markopoloparadox

I haven't done it. It's on my TODO list :)

**Review by Leouarz — APPROVED** [review]

**Review by ToufeeqP — APPROVED** [review]

**Pull request merged**